runtime/lua/vim/lsp.lua

```diff
@@ -635,6 +638,14 @@ function lsp.start(config, opts)
     config.root_dir = vim.fs.root(bufnr, opts._root_markers)
   end
 
+  if
+    not config.root_dir
+    and (not config.workspace_folders or #config.workspace_folders == 0)
+    and config.workspace_required
+  then
+    return
```

Comment:
please add a log message. silently doing nothing should be avoided.

please test that this works. then we're good to merge this PR.

```suggestion
    log.info(('skipping config "%s": workspace_required=true, and no workspace found'):format(config.name))
    return
```